nbi/engine/native/launcher/windows/src/JavaUtils.c

```diff
@@ -125,58 +125,79 @@ DWORD isJavaCompatible(JavaProperties *currentJava, JavaCompatible ** compatible
 JavaVersion * getJavaVersionFromString(char * string, DWORD * result) {
     JavaVersion *vers = NULL;
     if(getLengthA(string)>=3) {
-        //hope that at least we "major.minor" : 1.5
-        if(string[1]=='.') {
-            char c = string[0];
+        char *p = string;
```

Comment:
`const char*` is better, avoids accidental changes

The whole function body is guarded by `if(getLengthA(string)>=3)`; just immediately return `NULL` if `<3` and the number of nested `{` can decrease. Also handles `p == NULL` throughout the next code.

Reply:
Thanks for the hints!